pageserver/src/tenant.rs

```diff
@@ -459,6 +459,33 @@ struct RemoteStartupData {
     remote_metadata: TimelineMetadata,
 }
 
+#[derive(Debug)]
+pub(crate) enum AttachMarkerFilePresence {
+    Absent,
+    Present,
+}
+
+impl AttachMarkerFilePresence {
+    /// Returns that the claim made about marker file presence in `self`
+    /// corresponds to reality.
+    fn verify(
```

Comment:
Should this be called from somewhere?

Reply:
Depends on whether we have the `AttachMarkerFilePresence` at all.
Let's settle that discussion first, see in https://github.com/neondatabase/neon/pull/4168#discussion_r1186357400